sdk/python/kfp/dsl/_component_bridge.py

```diff
@@ -18,31 +18,86 @@
 from ..components._components import _default_component_name, _resolve_command_line_and_paths
 from ..components._naming import _sanitize_python_function_name, generate_unique_name_conversion_table
 from .. import dsl
-
+from kfp.ir import pipeline_spec_pb2
```

Comment:
You might want to fork the `_component_bridge`.
Currently the DSL compiler activates it during the compilation here: https://github.com/kubeflow/pipelines/blob/b1f46accdfeef67b2ff31e16e10304a6412f6b14/sdk/python/kfp/dsl/_pipeline.py#L234
(This should probably be moved here: https://github.com/kubeflow/pipelines/blob/b1f46accdfeef67b2ff31e16e10304a6412f6b14/sdk/python/kfp/compiler/compiler.py#L814)

You can do the same with the v2 compiler - activate the component bridge that will produce IR.

Reply:
Thanks. Are you suggesting forking only `_component_bridge.py` to something like `_component_bridge_v2.py`? I was considering whether or not to fork the whole dsl into v2/dsl. But for now it seems manageable to keep the change within v1 `_component_bridge.py` and make it compatible for both v1 and v2 behaviors. That said, forking may result better separation between v1 and v2 with cleaner v2 code. I might go with forking later.

I dont' quite follow the second part of your suggestion. Are you suggesting moving the shown code in `dsl/_pipeline.py` to `compiler/compiler.py`?

Reply:
>Are you suggesting forking only _component_bridge.py
Yes.

Although this is not a strong suggestion. There are pros and cons.

Reply:
forked _component_bridge.py